src/pixelmapFeature.js

```diff
@@ -0,0 +1,451 @@
+    m_info.indices = new Array(pixelData.length / 4);
+    m_info.area = pixelData.length / 4;
+
+    m_info.mappedColors = {};
+    for (i = 0; i < pixelData.length; i += 4) {
+      idx = pixelData[i] + (pixelData[i + 1] << 8) + (pixelData[i + 2] << 16);
+      m_info.indices[i / 4] = idx;
+      if (!m_info.mappedColors[idx]) {
+        m_info.mappedColors[idx] = {first: i / 4};
+      }
+      m_info.mappedColors[idx].last = i / 4;
+    }
+    return m_info;
+  };
+
+  /**
+   * Given the loaded pixelmap image, create a canvas the size of the image.
+   * Compute a color for each distinct index and recolor the canvas based on
+   * thise colors, then draw the resultant image as a quad.
+   */
+  this._computePixelmap = function () {
+    var data = m_this.data() || [],
+        mapColorFunc = m_this.style.get('mapColor'),
+        i, idx, lastidx, color, pixelData, indices, mappedColors,
+        updateFirst, updateLast = -1, update, prepared;
+
+    if (!m_info) {
+      if (!m_this._preparePixelmap()) {
+        return;
+      }
+      prepared = true;
+    }
+    mappedColors = m_info.mappedColors;
+    updateFirst = m_info.area;
+    for (idx in mappedColors) {
+      if (mappedColors.hasOwnProperty(idx)) {
+        color = mapColorFunc(data[idx], +idx) || {};
+        color = [
+          (color.r || 0) * 255,
+          (color.g || 0) * 255,
+          (color.b || 0) * 255,
+          color.a === undefined ? 255 : (color.a * 255)
+        ];
+        mappedColors[idx].update = (
+          !mappedColors[idx].color ||
+          mappedColors[idx].color[0] !== color[0] ||
+          mappedColors[idx].color[1] !== color[1] ||
+          mappedColors[idx].color[2] !== color[2] ||
+          mappedColors[idx].color[3] !== color[3]);
+        if (mappedColors[idx].update) {
+          mappedColors[idx].color = color;
+          updateFirst = Math.min(mappedColors[idx].first, updateFirst);
+          updateLast = Math.max(mappedColors[idx].last, updateLast);
+        }
+      }
+    }
+    /* If nothing was updated, we are done */
+    if (updateFirst >= updateLast) {
+      return;
+    }
+    /* Update only the extent that has changed */
+    pixelData = m_info.imageData.data;
+    indices = m_info.indices;
+    for (i = updateFirst; i <= updateLast; i += 1) {
+      idx = indices[i];
+      if (idx !== lastidx) {
+        lastidx = idx;
+        color = mappedColors[idx].color;
+        update = mappedColors[idx].update;
+      }
+      if (update) {
+        pixelData[i * 4] = color[0];
+        pixelData[i * 4 + 1] = color[1];
+        pixelData[i * 4 + 2] = color[2];
+        pixelData[i * 4 + 3] = color[3];
+      }
+    }
+    /* Place the updated area into the canvas */
+    m_info.context.putImageData(
+      m_info.imageData, 0, 0, 0, Math.floor(updateFirst / m_info.width),
+      m_info.width, Math.ceil((updateLast + 1) / m_info.width));
+
+    /* If we haven't made a quad feature, make one now.  The quad feature needs
```

Comment:
I am fine with that if we are consistent, may be we can change them at some point but i think right now we use single space. Not a roadblock for this PR though.